NitroxModel/DataStructures/Optional.cs

```diff
@@ -1,106 +1,101 @@
 ﻿using System;
+using ProtoBufNet;
 
 namespace NitroxModel.DataStructures.Util
 {
     [Serializable]
-    public class HasValueOptional<T> : Optional<T>
+    [ProtoContract]
+    public class Optional<T>
     {
-        private readonly T value;
-
-        public HasValueOptional(T value)
-        {
-            this.value = value;
+        public T Value
```

Comment:
What is the benefit of attribute vs function?  Is the switch worth nuking all of our easy access to git blame?

Reply:
In our previous implementation we had to convert from an Optional to a non optional value which may or may not fall apart at runtime(Proto issues)

Reply:
Also yea I should have left .Get() instead of just changing everything like this

Reply:
All other reasoning aside, the "rename" changes can be moved to a separate commit and listed in a rev-ignore file. Pass the file to [`git blame --ignore-revs-file`](https://git-scm.com/docs/git-blame#Documentation/git-blame.txt---ignore-revs-fileltfilegt) or set `blame.ignoreRevsFile` to the filename.

Granted, everyone needs to do this on their own, locally.